src/server/common/connectors/file-upload/cdp-uploader.test.js

```diff
@@ -0,0 +1,145 @@
+const mockBucket = 'mock-bucket'
+const mockPath = 'mock-path'
+const mockRedirectUrl = 'http://mock-redirect-url'
+const mockUploadId = 'test-upload-id'
+const mockUploadUrl = 'test-upload-url'
+const mockStatusUrl = 'test-status-url'
+
+const mockUploadConfig = {
+  bucket: mockBucket,
+  uploaderUrl: mockUploaderUrl,
+  path: mockPath
+}
+
+describe('uploadFile', () => {
+  beforeEach(() => {
+    config.get = jest.fn().mockImplementation((key) => {
+      if (key === 'fileUpload') {
+        return mockUploadConfig
+      }
+      if (key === 'featureFlags') {
+        return { pdfUpload: false }
+      }
+    })
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('should call Wreck.post with the correct payload and URL', async () => {
+    const wreckSpy = jest.spyOn(Wreck, 'post').mockResolvedValue({
+      res: /** @type {IncomingMessage} */ ({
+        statusCode: 200
+      }),
+      payload: JSON.stringify({
+        uploadId: mockUploadId,
+        uploadUrl: mockUploadUrl,
+        statusUrl: mockStatusUrl
+      })
+    })
+
+    await uploadFile(mockRedirectUrl)
+
+    expect(wreckSpy).toHaveBeenCalledWith(`${mockUploaderUrl}/initiate`, {
+      payload: JSON.stringify({
+        redirect: mockRedirectUrl,
+        s3Bucket: mockBucket,
+        s3Path: mockPath,
+        mimeTypes: ['image/png', 'image/jpeg'],
+        maxFileSize
+      })
+    })
+  })
+
+  it('should throw an error if Wreck.post fails', async () => {
+    const mockError = new Error('post failed')
+
+    jest.spyOn(Wreck, 'post').mockRejectedValue(mockError)
+
+    await expect(uploadFile(mockRedirectUrl)).rejects.toThrow('post failed')
+  })
+
+  describe('when pdfUpload feature flag is enabled', () => {
+    it('should include application/pdf in mimeTypes if pdfUpload feature flag is enabled', async () => {
+      config.get = jest.fn().mockImplementation((key) => {
+        if (key === 'fileUpload') {
+          return mockUploadConfig
+        }
+        if (key === 'featureFlags') {
+          return { pdfUpload: true }
+        }
+      })
+
+      const wreckSpy = jest.spyOn(Wreck, 'post').mockResolvedValue({
+        res: /** @type {IncomingMessage} */ ({
+          statusCode: 200
+        }),
+        payload: JSON.stringify({
+          uploadId: mockUploadId,
+          uploadUrl: mockUploadUrl,
+          statusUrl: mockStatusUrl
+        })
+      })
+
+      await uploadFile(mockRedirectUrl)
+
+      expect(wreckSpy).toHaveBeenCalledWith(`${mockUploaderUrl}/initiate`, {
+        payload: JSON.stringify({
+          redirect: mockRedirectUrl,
+          s3Bucket: mockBucket,
+          s3Path: mockPath,
+          mimeTypes: ['image/png', 'image/jpeg', 'application/pdf'],
+          maxFileSize
+        })
+      })
+    })
+  })
+})
+
+describe('checkStatus', () => {
+  beforeEach(() => {
+    spyOnConfig('fileUpload', mockUploadConfig)
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
```

Comment:
jest.restoreAllMocks() preferred. 

Also, it can just be `afterEach(jest.restoreAllMocks)`, FWIW.